graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/execution/FederatedTypeResolver.kt

```diff
@@ -23,6 +23,12 @@ import graphql.schema.DataFetchingEnvironment
  */
 interface FederatedTypeResolver<out T> {
 
+    /**
+     * This is the GraphQL name of the type [T]. It is used when running the resolvers and inspecting the
+     * GraphQL "__typename" property during the entities requests
+     */
+    val typeName: String
```

Comment:
guess this is no different than having to specify explicit `string` -> resolver mapping. Ideally this should be coming from the underlying type `T` but when I last looked at it was problematic due to type erasure.

Reply:
Yea, this just removes the needed extra step that you need to create a map of `Strings` to resolvers. I think it is more clear here that the type name should be part of the resolver so you only have to change code in one location